centreon/src/Core/ServiceTemplate/Infrastructure/Repository/DbReadServiceTemplateRepository.php

```diff
@@ -0,0 +1,290 @@
+            }
+
+            return $commandArguments;
+        };
+
+        return new ServiceTemplate(
+            (int) $data['service_id'],
+            $data['service_description'],
+            $data['service_alias'],
+            $extractCommandArgument($data['command_command_id_arg']),
+            $extractCommandArgument($data['command_command_id_arg2']),
+            $this->createNotificationType($data['service_notification_options']),
+            $data['contact_additive_inheritance'] === 1,
+            $data['cg_additive_inheritance'] === 1,
+            $data['service_activate'] === '1',
+            $data['service_locked'] === 1,
+            $this->createYesNoDefault($data['service_active_checks_enabled']),
+            $this->createYesNoDefault($data['service_passive_checks_enabled']),
+            $this->createYesNoDefault($data['service_is_volatile']),
+            $this->createYesNoDefault($data['service_check_freshness']),
+            $this->createYesNoDefault($data['service_event_handler_enabled']),
+            $this->createYesNoDefault($data['service_flap_detection_enabled']),
+            $this->createYesNoDefault($data['service_notifications_enabled']),
+            $data['service_description'],
+            $data['service_comment'],
+            $data['esi_notes'],
+            $data['esi_notes_url'],
+            $data['esi_action_url'],
+            $data['esi_icon_image_alt'],
+            $data['graph_id'],
+            $data['service_template_model_stm_id'],
+            $data['command_command_id'],
+            $data['command_command_id2'],
+            $data['timeperiod_tp_id2'],
+            $data['timeperiod_tp_id'],
+            $data['esi_icon_image'],
+            $data['severity_id'],
+            $data['service_max_check_attempts'],
+            $data['service_normal_check_interval'],
+            $data['service_retry_check_interval'],
+            $data['service_freshness_threshold'],
+            $data['service_low_flap_threshold'],
+            $data['service_high_flap_threshold'],
+            $data['service_notification_interval'],
+            $data['service_recovery_notification_delay'],
+            $data['service_first_notification_delay'],
+            $data['service_acknowledgement_timeout']
+        );
+    }
+
+    private function createYesNoDefault(string $value): YesNoDefault
+    {
+        return match ($value) {
+            '0' => YesNoDefault::No,
+            '1' => YesNoDefault::Yes,
+            default => YesNoDefault::Default
+        };
+    }
+
+    /**
+     * @param string|null $notificationTypes
+     *
+     * @return NotificationType[]
+     */
+    private function createNotificationType(?string $notificationTypes): array
+    {
+        if ($notificationTypes === null) {
+            return [];
+        }
+        $notifications = [];
+
+        $types = preg_split('|,|', $notificationTypes);
+        if (is_array($types)) {
+            foreach ($types as $type) {
```

Comment:
We don't have a redundancy check like `aray_unique`.
```suggestion
            foreach (array_unique($types) as $type) {
```